scanpy/settings.py

```diff
@@ -51,6 +52,10 @@
 """Default cache directory.
 """
 
+datasetdir = Path.home() / "scanpy_datasets"
```

Comment:
Veto, see #558. I consider creating `~/something-visible` a completely unacceptable default for a library.

Please leave this change out of this PR. We can do a separate one once we have decided on how to tackle this.

Reply:
Sure, I also think it would be a good idea to leave changing the default to a separate PR.

You'll see I initially made this a non-breaking change (`datasetdir = Path("./data")`), just changed to this going off feedback from Alex.

I'm gonna wait on actually modifying the code for a minute to make the eventual rebase easier.

Reply:
OK, great. I think the changes to the default (`./data` → `~/???`) should be in a PR that also changes what’s printed on download/read of such files, and we can figure out there how to detect if we’re on a linux server (where `~/scanpy_data` would make sense) or on a desktop system (where it doesn’t)

Reply:
Would you mind sharing some links which explain why this is important? I'll admit, it does not seem very complex to me at the moment, but I also don't know much about this topic.

Also, a more general question, does scanpy work on windows?

Reply:
Sadly there’s no writeup. It’s a lot of different observations I made and discussions in different projects. If you search XDG_CONFIG_HOME and [XDG_CACHE_HOME](https://github.com/search?q=xdg_cache_home&type=Issues) you’ll find a lot of people requesting that (e.g. mwaskom/seaborn#1530)

The question “why not” is always answered with “we didn’t think about it” or “we followed what others did (without thinking about it)”, and the answer to “why don’t you change” is always “we have been doing it for too long, people will get confused”. So I won’t wait until somebody is able to say the latter about scanpy.

----

About Windows: Yes! I think it’s important to keep it that way. I recently read [this blog post](https://samsaffron.com/archive/2019/03/31/why-i-stuck-with-windows-for-6-years-while-developing-discourse) and fully agree with the author:

> […] a Ruby that works well on native Windows will help Ruby adoption a lot and eliminate drain to other frameworks. Installing a whole brand new OS is just too much of a barrier. Just install Linux is not a solution.

The same applies to everything of course. Many just start out and aren’t fed up by Windows enough to ditch it, so making the start on Windows as smooth as possible helps adoption by new people a lot.

Reply:
I'll check out that issue.

I have nothing against windows support, I just can't help on any issues related to it. I was mostly wondering if all the dependencies worked on windows, and if there's a reason there aren't appveyor builds.

Reply:
This requested change is resolved now, right?